src/module.ts

```diff
@@ -1081,14 +1081,100 @@ export const enum RefAsOp {
+  BrOnNonFunc = 5 /* TODO_BinaryenBrOnNonFunc */,
   /** br_on_data */
-  BrOnData = 3 /* TODO_BinaryenBrOnData */,
+  BrOnData = 6 /* TODO_BinaryenBrOnData */,
+  /** br_on_non_data */
+  BrOnNonData = 7 /* TODO_BinaryenBrOnNonData */,
   /** br_on_i31 */
-  BrOnI31 = 4 /* TODO_BinaryenBrOnI31 */
+  BrOnI31 = 8 /* TODO_BinaryenBrOnI31 */,
+  /** br_on_non_i32 */
+  BrOnNonI31 = 9 /* TODO_BinaryenBrOnNonI31 */
+}
+
+/** Binaryen StringNew operation constants. */
+export const enum StringNewOp {
+  /** string.new_wtf8 utf8 */
+  StringNewUTF8 = 0 /* TODO_BinaryenStringNewUTF8 */,
+  /** string.new_wtf8 wtf8 */
+  StringNewWTF8 = 1 /* TODO_BinaryenStringNewWTF8 */,
+  /** string.new_wtf8 replace */
+  StringNewReplace = 2 /* TODO_BinaryenStringNewReplace */,
+  /** string.new_wtf16 */
+  StringNewWTF16 = 3/* TODO_BinaryenStringNewWTF16 */,
+  /** string.new_wtf8_array utf8 */
+  StringNewUTF8Array = 4 /* TODO_BinaryenStringNewUTF8Array */,
+  /** string.new_wtf8_array wtf8 */
+  StringNewWTF8Array = 5 /* TODO_BinaryenStringNewWTF8Array */,
+  /** string.new_wtf8_array replace */
+  StringNewReplaceArray = 6 /* TODO_BinaryenStringNewReplaceArray */,
+  /** string.new_wtf16_array */
+  StringNewWTF16Array = 7 /* TODO_BinaryenStringNewUTF16Array */
+}
+
+/** Binaryen StringMeasure operation constants. */
+export const enum StringMeasureOp {
+  /** string.measure_wtf8 utf8 */
+  StringMeasureUTF8 = 0 /* TODO_BinaryenStringMeastureUTF8 */,
+  /** string.measure_wtf8 wtf8 */
+  StringMeasureWTF8 = 1 /* TODO_BinaryenStringMeastureWTF8 */,
+  /** string.measure_wtf16 */
+  StringMeasureWTF16 = 2 /* TODO_BinaryenStringMeastureWTF16 */,
+  /** string.is_usv_sequence */
+  StringMeasureIsUSV = 3 /* TODO_BinaryenStringMeasureIsUSV */,
+  /** stringview_wtf16.length */
+  StringMeasureWTF16View = 4 /* TODO_BinaryenStringMeasureWTF16View */
+}
+
+/** Binaryen StringEncode operation constants. */
+export const enum StringEncodeOp {
+  /** string.encode_wtf8 utf8 */
+  StringEncodeUTF8 = 0 /* TODO_BinaryenStringEncodeUTF8 */,
+  /** string.encode_wtf8 wtf8 */
+  StringEncodeWTF8 = 1 /* TODO_BinaryenStringEncodeWTF8 */,
+  /** string.encode_wtf16 */
+  StringEncodeWTF16 = 2 /* TODO_BinaryenStringEncodeWTF16 */,
+  /** string.encode_wtf8_array utf8 */
+  StringEncodeUTF8Array = 3 /* TODO_BinaryenStringEncodeUTF8Array */,
+  /** string.encode_wtf8_array wtf8 */
+  StringEncodeWTF8Array = 4 /* TODO_BinaryenStringEncodeWTF8Array */,
+  /** string.encode_wtf16_array */
+  StringEncodeWTF16Array = 5 /* TODO_BinaryenStringEncodeWTF16Array */
+}
+
+/** Binaryen StringAs operation constants. */
+export const enum StringAsOp {
+  /** string.as_wtf8 */
+  StringAsWTF8 = 0 /* TODO_BinaryenStringAsWTF8 */,
+  /** string.as_wtf16 */
+  StringAsWTF16 = 1 /* TODO_BinaryenStringAsWTF16 */,
+  /** string.as_iter */
+  StringAsIter = 2 /* TODO_BinaryenStringAsIter */
+}
+
+/** Binaryen StringIterMove operation constants. */
+export const enum StringIterMoveOp {
```

Comment:
How about just `StringIterOp`?

Reply:
So far these are 1:1 copies of what Binaryen has, where there is a `StringIterMove` instruction class taking these constants as their `op` field.

Reply:
ok